pygsti/circuits/circuit.py

```diff
@@ -927,6 +909,33 @@ def __add__(self, x):
 
         return Circuit._fastinit(self.layertup + x.layertup, new_line_labels, editable=False, name='',
                                  stringrep=s, occurrence=None)
+    
+
+    def sandwich(self, x, y):
```

Comment:
This is used in the implementation of the new `complete_circuits` method in `OpModel`. I got to a point in optimizing that where the runtime was dominated by the two circuit additions with prepend and postpend the prep and POVM labels (~80% of runtime was just that). It then occurred to me that since almost all of the time for `__add__` is the `_fastinit` call that we could do the sandwiching operation in nearly the same time as a single addition. Even with that change the `sandwich` calls still account for ~50% of that method's runtime.